### PR TITLE
bots: Add machine learning for test flakes and  failures

### PR DESCRIPTION
**Pull request by stefwalter**
Goal is to start putting numbers under each test failure whether we think this is a flake or not.

 * [x] #7403 
 * [x] #7808 
 * [x] https://github.com/cockpit-project/cockpituous/pull/152
 * [x] tests-learn script
 * [x] tests-policy script

### CODE REVIEW

**Comment by stefwalter** [comment]
Rebased.

**Comment by martinpitt** [comment]
All dependencies landed, but needs a rebase now.

**Comment by stefwalter** [comment]
Rebased

**Comment by stefwalter** [comment]
This is not perfect yet, but it's starting to work:

```
# testTeam (check_networking_team.TestNetworking)
#
Warning: Permanently added '[127.0.0.2]:2601' (ECDSA) to the list of known hosts.
Warning: Permanently added '[127.0.0.2]:2602' (ECDSA) to the list of known hosts.
NAME         UUID                                  TYPE            DEVICE 
System eth0  5fb06bd0-0bb0-7ffb-45f1-d6edd65f3e03  802-3-ethernet  eth0   
virbr0       24919ce5-8197-48f5-a08a-581f8274f04c  bridge          virbr0 
System eth1  9c92fad9-6ecb-3e6c-eb4d-8a47c6f50c04  802-3-ethernet  --     

52:54:01:00:00:03 -> eth2
52:54:01:00:00:03 -> eth2
52:54:01:00:00:04 -> eth3
52:54:01:00:00:04 -> eth3
Unexpected journal message '/usr/libexec/cockpit-pcp: bridge was killed: 11'
Journal extracted to TestNetworking-testTeam-rhel-7-4-127.0.0.2-2601-FAIL.log
Journal extracted to TestNetworking-testTeam-fedora-26-127.0.0.2-2602-FAIL.log
Core dumps downloaded to /build/cockpit/TestNetworking-testTeam-rhel-7-4-127.0.0.2-2601-FAIL.core
Traceback (most recent call last):
  File "/build/cockpit/test/common/testlib.py", line 602, in tearDown
    self.check_journal_messages()
  File "/build/cockpit/test/common/testlib.py", line 773, in check_journal_messages
    raise Error(first)
Error: /usr/libexec/cockpit-pcp: bridge was killed: 11

not ok 35 testTeam (check_networking_team.TestNetworking) # duration: 50s
Wrote TestNetworking-testTeam-rhel-7-4-127.0.0.2-2601-FAIL.png
Wrote TestNetworking-testTeam-rhel-7-4-127.0.0.2-2601-FAIL.html
Journal extracted to TestNetworking-testTeam-rhel-7-4-127.0.0.2-2601-FAIL.log
Journal extracted to TestNetworking-testTeam-fedora-26-127.0.0.2-2602-FAIL.log
Core dumps downloaded to /build/cockpit/TestNetworking-testTeam-rhel-7-4-127.0.0.2-2601-FAIL.core


# Flake probability: 70.0%
# Flake debug: array([ 0.2995244,  0.7004756])
```

File: bots/tests-policy [hunk 1]
```diff
@@ -111,6 +121,55 @@ def filterRetry(output, prefix):
...
+        "date": time.strftime("%Y-%m-%dT%H:%M:%SZ"),
+        "merged": None,
+        "test": parseName(output),
+        "url": None,
+        "log": output
+    }
+
+    pred_proba = network.predict_proba(item)
+    if max(pred_proba) >= 0.5:
+        if int(pred_proba[1] >= pred_proba[0]) == 1:
```
Comment (martinpitt) [inline bots/tests-policy]: The `int(…) == 1` should be replaced with `…`, but with a construct like that it's actually likely that the author meant something entirely different.

**Pull request merged**